### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Sums.Basic

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
Might need another look to check for names4 conformity.

File: Mathlib/CategoryTheory/Sums/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+#align category_theory.sum.swap_map_inl CategoryTheory.Sum.swap_map_inl
+
+@[simp]
+theorem swap_map_inr {X Y : D} {f : inr X ⟶ inr Y} : (swap C D).map f = f :=
+  rfl
+#align category_theory.sum.swap_map_inr CategoryTheory.Sum.swap_map_inr
+
+namespace Swap
+
+/- Porting note: had ot manually call `cases f` for `f : PEmpty` -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Sums/Basic.lean:144]:
```suggestion
/- Porting note: had to manually call `cases f` for `f : PEmpty` -/
```

Comment (jcommelin) [inline Mathlib/CategoryTheory/Sums/Basic.lean:144]:
```suggestion
/- Porting note: had to manually call `cases f` for `f : PEmpty` -/
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**